core/src/main/java/org/everit/email/store/ri/EmailStoreImpl.java

```diff
@@ -0,0 +1,688 @@
+          .set(qHtmlContent.blobId, htmlContentBlobId)
+          .set(qHtmlContent.storedEmailId, storedEmailId)
+          .executeWithKey(qHtmlContent.htmlContentId);
+    });
+    insertInlineImages(htmlContent.inlineImageByCidMap, htmlContentId);
+  }
+
+  private void insertInlineImages(final Map<String, Attachment> inlineImageByCidMap,
+      final long htmlContentId) {
+    if (inlineImageByCidMap == null) {
+      return;
+    }
+
+    querydslSupport.execute((connection, configuration) -> {
+      int index = START_INDEX;
+      for (Map.Entry<String, Attachment> entry : inlineImageByCidMap.entrySet()) {
+        Long binaryContentId = insertBinaryContent(entry.getValue());
+        QInlineImage qInlineImage = QInlineImage.inlineImage;
+        new SQLInsertClause(connection, configuration, qInlineImage)
+            .set(qInlineImage.cid_, entry.getKey())
+            .set(qInlineImage.binaryContentId, binaryContentId)
+            .set(qInlineImage.index_, index++)
+            .set(qInlineImage.htmlContentId, htmlContentId)
+            .executeWithKey(qInlineImage.inlineImageId);
+      }
+      return null;
+    });
+  }
+
+  private void insertRecipient(final EmailAddress emailAddress, final RecipientType recipientType,
+      final int index, final long storedEmailId) {
+    long emailAddressId = insertAddress(emailAddress);
+
+    querydslSupport.execute((connection, configuration) -> {
+      QEmailRecipient qEmailRecipient = QEmailRecipient.emailRecipient;
+      return new SQLInsertClause(connection, configuration, qEmailRecipient)
+          .set(qEmailRecipient.emailAddressId, emailAddressId)
+          .set(qEmailRecipient.index_, index)
+          .set(qEmailRecipient.storedEmailId, storedEmailId)
+          .set(qEmailRecipient.recipientType, recipientType.name())
+          .executeWithKey(qEmailRecipient.emailRecipientId);
+    });
+  }
+
+  private void insertRecipients(final Collection<EmailAddress> emailAddresses,
+      final RecipientType recipientType, final long storedEmailId) {
+    if (emailAddresses == null) {
+      return;
+    }
+
+    int index = START_INDEX;
+    for (EmailAddress emailAddress : emailAddresses) {
+      insertRecipient(emailAddress, recipientType, index++, storedEmailId);
+    }
+  }
+
+  private void insertTextContent(final String textContent, final long storedEmailId) {
+    if (textContent == null) {
+      return;
+    }
+
+    Long textContentBlobId = createBlob(getBytes(textContent));
+    querydslSupport.execute((connection, configuration) -> {
+      QTextContent qTextContent = QTextContent.textContent;
+      return new SQLInsertClause(connection, configuration, qTextContent)
+          .set(qTextContent.storedEmailId, storedEmailId)
+          .set(qTextContent.blobId, textContentBlobId)
+          .executeWithKey(qTextContent.textContentId);
+    });
+  }
+
+  private boolean isEmailExists(final long storedEmailId) {
```

Comment:
We want to lock the email we want to delete with a forUpdate() so two parallel threads will not cause troubles. I would change the function name, too: lockEmailForUpdate and the function would throw an NonExistentEmailException if the email does not exist.